docs/proposals/app-metrics.md

```diff
@@ -0,0 +1,124 @@
+# Application metrics
+ 
+## Context
+ 
+Currently, when an application exposes metrics as regular service it's protected and traffic is secured (mTLS enabled), it's a bit tricky
+to [expose those metrics without mTLS](https://kuma.io/docs/dev/policies/traffic-metrics/#expose-metrics-from-applications). Also, by configuring paths user's can by mistake expose more than it's necessary. To avoid it we should come up with a solution that fulfills all the requirements: scraping metrics with/without mTLS and safer exposing of metrics.
+ 
+## Requirements
+* support different application ports/paths
+* support more than one container/application
+* support no mTLS communication with Prometheus
+ 
+## Idea
+ 
+The idea is to create an aggregator in `kuma-dp`, to be specific Metrics Hijacker, that will be responsible for the aggregation of metrics from applications defined to scrap(within local env: pod/vm). Communication with `kuma-dp` might be configured as secure but it won't be required(and not recommended). It's worth mentioning that this idea can support only Prometheus endpoint scrapping and exposing metrics as Prometheus.
+ 
+### Universal
+At the mesh level, it will be possible to define all applications endpoints that should be scrapped by `kuma-dp` within a Mesh.
+```
+type: Mesh
+name: default
+metrics:
+ enabledBackend: prometheus-1
+ backends:
+ - name: prometheus-1
+   type: prometheus
+   conf:
+     skipMTLS: true
+     mtls:
+       require: true
+       tags: # tags that can be referred in Traffic Permission when metrics are secured by mTLS 
+         kuma.io/service: dataplane-metrics
+     aggregate:
+     - name: app
+       port: 1234
+       path: /metrics # defaults to /metrics
+     - name: opa
+       port: 8888
+       path: /metrics # defaults to /metrics
+```
+`Dataplane` configuration enables at which port/path `kuma-dp` is going to expose aggregated metrics from Envoy and other applications running within one workflow. Also, it's possible to override the `Mesh` configuration of endpoints to scrap by the `Dataplane` configuration. They are going to be identified by `name` and if `Mesh` and `Dataplane` configurations have definitions with the same name, then the `Dataplane` configuration has precedence before `Mesh` and is merged.
+```
+type: Dataplane
+mesh: default
+name: redis-1
+networking:
+ address: 192.168.0.1
+ inbound:
+ - port: 9000
+   servicePort: 1234
+   tags:
+     kuma.io/service: backend
+ metrics:
+   type: prometheus
+   conf:
+     skipMTLS: true
+     port: 1234
+     path: /non-standard-path
```

Comment:
Should this be a DP option?

Reply:
I think for now it's going to be easier to add data to dataplane object. Creating a new policy would be probably better but we also working on new policies so it might be blocked for a while.